File: Client/Linux/lib/Reporting/ReportingClient.cpp

#include "ReportingClient.h"
#include "../RestConnection.h"

using namespace std;

namespace Thriot { namespace Reporting {

/**
Craete a new instance of the Reporting client

@param baseUrl Base Reporting API url
*/
ReportingClient::ReportingClient(const string& baseUrl)
{
	_restConnection = new RestConnection(baseUrl);
	_deviceClient = new DeviceClient(_restConnection);
	_networkClient = new NetworkClient(_restConnection);
}

/** Cleanup the instance */
ReportingClient::~ReportingClient()
{
	delete _networkClient;
	delete _deviceClient;
	delete _restConnection;
}


/**
Return the device-level reporting client

@return */
DeviceClient* ReportingClient::Device()
{
	return _deviceClient;
}

/**
Return the network-level reporting client

@return */
NetworkClient* ReportingClient::Network()
{
	return _networkClient;
}
}}

File: Client/DotNet/IoT.Client.DotNet/Reporting/ReportingClient.cs
namespace IoT.Client.DotNet.Reporting
{
    /// <summary>
    /// Umbrella class for device and network based reporting functions.
    /// The reporting functions are suitable for creating JSON and CSV output based on the recorded telemetry data of single or multiple devices.
    /// </summary>
    public class ReportingClient
    {
        /// <summary>
        /// Creates a new instance of reporting client
        /// </summary>
        /// <param name="baseUrl">Reporting API root</param>
        public ReportingClient(string baseUrl) : this(baseUrl, new RestConnection())
        {
        }

        private ReportingClient(string baseUrl, IRestConnection restConnection)
        {
            restConnection.Setup(baseUrl, null);

            Device = new DeviceClient(baseUrl, restConnection);
            Network = new NetworkClient(baseUrl, restConnection);
        }

        /// <summary>
        /// Device-specific reporting functions
        /// </summary>
        public DeviceClient Device { get; private set; }

        /// <summary>
        /// Network-specific reporting functions
        /// </summary>
        public NetworkClient Network { get; private set; }
    }
}
